ipaclient/plugins/certmapping.py

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright (C) 2016  FreeIPA Contributors see COPYING for license
+#
+
+from ipalib import api
+from ipalib import errors
+from ipalib import output
+from ipalib import util
+from ipalib.certmapping import CSRGenerator, FileRuleProvider
+from ipalib.frontend import Command, Str
+from ipalib.parameters import Principal
+from ipalib.plugable import Registry
+from ipalib.text import _
+from ipaserver.plugins.certprofile import validate_profile_id
+
+import six
+
+if six.PY3:
+    unicode = str
+
+register = Registry()
+
+__doc__ = _("""
+Commands to build certificate requests automatically
+""")
+
+
+@register()
+class cert_get_requestdata(Command):
+    __doc__ = _('Gather data for a certificate signing request.')
+
+    takes_options = (
+        Principal(
+            'principal',
+            label=_('Principal'),
+            doc=_('Principal for this certificate (e.g.'
+                  ' HTTP/test.example.com)'),
+        ),
+        Str(
+            'profile_id',
+            validate_profile_id,
+            label=_('Profile ID'),
+            doc=_('Certificate Profile to use'),
+        ),
+        Str(
+            'helper',
+            label=_('Name of CSR generation tool'),
+            doc=_('Name of tool (e.g. openssl, certutil) that will be used to'
+                  ' create CSR'),
+        ),
+        Str(
+            'out?',
+            doc=_('Write CSR generation script to file'),
+        ),
+    )
+
+    has_output = (
+        output.Output(
+            'result',
+            type=dict,
+            doc=_('Dictionary mapping variable name to value'),
+        ),
+    )
+
+    has_output_params = (
+        Str(
+            'script',
+            label=_('Generation script'),
+        )
+    )
+
+    def forward(self, *args, **options):
```

Comment:
Ah, I had tried that before but abandoned it because the rpc client wasn't connected automatically with `Local`. But now I see that `vault-add` connects it explicitly, so I've updated my code to do the same.